pertpy/tools/_perturbation_space/_clustering.py

```diff
@@ -1,14 +1,128 @@
+    ):
+        """Initialize ClusteringSpace class.
+
+        Args:
+            method: ClusteringSpace method to use. Defaults to 'K-Means'.
+        """
+
+        if method.lower() == "k-means":
+            from pertpy.tools._perturbation_space._simple import KMeansSpace
+
+            method_fct = KMeansSpace()  # type: ignore
+        elif method.lower() == "dbscan":
+            from pertpy.tools._perturbation_space._simple import DBScanSpace
+
+            method_fct = DBScanSpace()  # type: ignore
+        elif method.lower() == "pseudobulk":
+            from pertpy.tools._perturbation_space._simple import PseudobulkSpace
+
+            method_fct = PseudobulkSpace()  # type: ignore
+        elif method.lower() == "classifier":
+            from pertpy.tools._perturbation_space._discriminator_classifier import DiscriminatorClassifierSpace
+
+            method_fct = DiscriminatorClassifierSpace()  # type: ignore
+        else:
+            raise ValueError(f"Method {method} not recognized.")
+        self.method_fct = method_fct
+
+    def __call__(  # type: ignore
         self,
-        *args,
-        clustering_fct: Literal["dendrogram", "k-means", "dbscan", "spectral clustering", "gaussian mixture"],
-        **kwargs
+        adata: AnnData,
+        *,
+        reference_key: str = "control",
+        layer_key: str = None,
+        new_layer_key: str = None,
+        embedding_key: str = None,
+        new_embedding_key: str = None,
+        copy: bool = False,
+        **kwargs,
+    ) -> AnnData:
+        """Compute clustering space.
+
+        Args:
+            adata: The AnnData object that contains the data.
+            layer_key: The layer of which to use the transcription values for to determine the perturbation space.
+            embedding_key: The obsm matrix of which to use the embedding values for to determine the perturbation space.
+
+        Returns:
+            The determined perturbation space inside an AnnData object.
+            # TODO add more detail
+
+        Examples:
+            >>> import pertpy as pt
+
+            >>> adata = pt.dt.papalexi_2021()["rna"]
+            >>> clsp = pt.tl.ClusteringSpace(method='K-Means')
+            >>> perturbation_space = clsp()
+            >>> TODO add something here
+        """
+        method_kwargs = {
+            "reference_key": reference_key,
+            "layer_key": layer_key,
+            "new_layer_key": new_layer_key,
+            "embedding_key": embedding_key,
+            "new_embedding_key": new_embedding_key,
+            "copy": copy,
+            **kwargs,
+        }
+
+        return self.method_fct(adata, **method_kwargs)  # type: ignore
+
+    def evaluate(
+        self,
+        adata: AnnData,
+        true_label_col: str,
+        cluster_col: str,
+        metrics: List[str] = None,
+        **kwargs,
     ):
-        # TODO implement
-        pass
+        """Evaluation of previously computed clustering against ground truth labels
```

Comment:
This sounds like a pretty general function. Should this be in `_perturbation_space.py`?

Reply:
as mentioned, now `ClusteringSpace` is different from `PerturbationSpace`. Nevertheless, maybe we can incorporate it in the `PerturbationSpace` specially for the case of the `DiscriminatorClassifierSpace` (whose name I think we could maybe change...)
I let this open to discuss it

Reply:
Yeah, I'd move it.